src/pipelinedefinition.hpp

```diff
@@ -71,13 +73,37 @@ struct NodeInfo {
 };
 
 class PipelineDefinition {
+    struct ValidationResultNotifier {
+        ValidationResultNotifier(PipelineDefinitionStatus& status) :
+            status(status) {}
+        ~ValidationResultNotifier() {
+            if (passed) {
+                status.handle(ValidationPassedEvent());
+            } else {
+                status.handle(ValidationFailedEvent());
+            }
+        }
+        bool passed = false;
+
+    private:
+        PipelineDefinitionStatus& status;
+    };
+
     const std::string pipelineName;
     std::vector<NodeInfo> nodeInfos;
     pipeline_connections_t connections;
-    std::set<std::pair<const std::string, model_version_t>> subscriptions;
+
     std::atomic<uint64_t> requestsHandlesCounter = 0;
     std::shared_mutex loadMtx;
 
+    std::condition_variable loadedNotify;
+
+protected:
+    PipelineDefinitionStatus status;
```

Comment:
Why is this field protected? Is it only to expose this field for unit tests? What about mocking getStatus method like we do with ModelInstance?

Reply:
PipelineDefinitionStatus(PDS) shouldn't be controllable from outside of PipelineDefinition(PD). PipelineDefinition should just forward the appropriate event to its machine state - PDS. I think we should give control over PDS to external stakeholder than PD.

I assume PD interface will give const copy of PDS/const ref or just the StateCode.